src/relations.jl

```diff
@@ -1320,22 +1320,27 @@ is a function that is 1 above `T_freeze` and goes to zero below `T_icenuc`.
     T::FT,
     ::Type{phase_type},
     q::PhasePartition{FT} = q_pt_0(FT),
-) where {FT <: Real, phase_type <: ThermodynamicState}
-    _T_freeze::FT = TP.T_freeze(param_set)
-    _T_icenuc::FT = TP.T_icenuc(param_set)
-
-    if T > _T_freeze
-        return FT(1)
-    elseif T > _T_icenuc
-        _pow_icenuc::FT = TP.pow_icenuc(param_set)
-        if _pow_icenuc == 1
-            return (T - _T_icenuc) / (_T_freeze - _T_icenuc)
-        else
-            return ((T - _T_icenuc) / (_T_freeze - _T_icenuc))^_pow_icenuc
-        end
-    else
-        return FT(0)
-    end
+) where {FT, phase_type <: ThermodynamicState}
+
+    Tᶠ = TP.T_freeze(param_set)   # freezing temperature
+    Tⁿ = TP.T_icenuc(param_set)   # temperature of homogeneous ice nucleation
+    α = TP.pow_icenuc(param_set) # power law partial ice nucleation parameter
+
+    # Model for cloud water condensate probabilty when temperature is below
+    # freezing (all liquid condensate), but above the temperature of homogeneous
+    # ice nucleation (all ice condensate). In it's simplest form, this is simply
+    # a number that varies between 0 and 1. For example, see figure 6 in 
+    # Hu et al., https://doi.org/10.1029/2009JD012384, JGR (2010).
+    λᵖ = ((T - Tⁿ) / (Tᶠ - Tⁿ))^α
```

Comment:
One reason that I introduced branching was to avoid calling the generic `pow` (which is quite expensive) in the case when `α==1`. I suppose we can use one more level of `ifelse` nestedness to recover that optimization.

Reply:
When we use `ifelse`, both branches are computed. Thus no savings can be realized; adding a computation for the case `α==1` will always produce a slow down. This is a generic feature of GPU computing --- typically, all computations are performed at every grid point, and `if` statements predicated on runtime values cannot be used for optimization. It is possible of course that for extremely heavy code, that branching / `if` statements can produce a performance benefit. But one should not expect to use such constructs.

In this case, `α==1` can actually be known at compile time. The case `α isa Int` is also important. To express this important performance optimization, the parameter `pow_icenuc` should be allowed to be a type other than `FT` so the optimization can be resolved at compile time.

Reply:
I don't think we'll see much of any performance difference with removing the branch. I did see an improvement by including this branch on the CPU. `pow` is generally pretty expensive. Also, we previously ran into https://stackoverflow.com/questions/14687665/very-slow-stdpow-for-bases-very-close-to-1, which turned out to be extremely expensive and hence why we implemented `pow_hack`.